From 2e9c264489a2bc1176550d9a4cb90f26052dd5fd Mon Sep 17 00:00:00 2001
From: Peter Robinson <pbrobinson@gmail.com>
Date: Sat, 13 Jul 2019 00:43:52 +0100
Subject: [PATCH 2/4] Default to PulseAudio

Signed-off-by: Peter Robinson <pbrobinson@gmail.com>
---
 Top/csound.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Top/csound.c b/Top/csound.c
index d1e7fdc74..1663ac876 100644
--- a/Top/csound.c
+++ b/Top/csound.c
@@ -3452,7 +3452,7 @@ PUBLIC void csoundReset(CSOUND *csound)
       strcpy(s, "PortAudio");
  #endif
 #else
-    strcpy(s, "alsa");
+    strcpy(s, "pulse");
 #endif
     csoundCreateConfigurationVariable(csound, "rtaudio", s, CSOUNDCFG_STRING,
                                       0, NULL, &max_len,
-- 
2.21.0

