From fe0d1b46c74e451e6dbd6d19ca61d66d4e704594 Mon Sep 17 00:00:00 2001
From: Peter Robinson <pbrobinson@gmail.com>
Date: Thu, 1 Feb 2018 09:26:57 +0000
Subject: [PATCH 3/5] Default to PulseAudio

Signed-off-by: Peter Robinson <pbrobinson@gmail.com>
---
 Top/csound.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Top/csound.c b/Top/csound.c
index 9b95674df..bdf8529f8 100644
--- a/Top/csound.c
+++ b/Top/csound.c
@@ -3425,7 +3425,7 @@ PUBLIC void csoundReset(CSOUND *csound)
     csoundCreateGlobalVariable(csound, "_RTAUDIO", (size_t) max_len);
     s = csoundQueryGlobalVariable(csound, "_RTAUDIO");
 #ifndef LINUX
-    strcpy(s, "PortAudio");
+    strcpy(s, "pulse");
 #else
     strcpy(s, "alsa");
 #endif
-- 
2.19.0

