--- CMakeLists.txt.orig	2014-05-04 05:49:41.159716977 -0600
+++ CMakeLists.txt	2014-05-05 20:00:00.000000000 -0600
@@ -27,11 +27,7 @@ else()
 endif()
 message(STATUS "LIBRARY INSTALL DIR: ${LIBRARY_INSTALL_DIR}")
 
-if(USE_DOUBLE)
-    set(PLUGIN_INSTALL_DIR "${LIBRARY_INSTALL_DIR}/csound/plugins64-${APIVERSION}")
-else()
-    set(PLUGIN_INSTALL_DIR "${LIBRARY_INSTALL_DIR}/csound/plugins-${APIVERSION}")
-endif()
+set(PLUGIN_INSTALL_DIR "${LIBRARY_INSTALL_DIR}/csound/plugins-${APIVERSION}")
 set(PYTHON_MODULE_INSTALL_DIR ${LIBRARY_INSTALL_DIR})
 set(JAVA_MODULE_INSTALL_DIR ${LIBRARY_INSTALL_DIR})
 set(LUA_MODULE_INSTALL_DIR ${LIBRARY_INSTALL_DIR})
--- installer/misc/mkpackage.py.orig	2014-05-04 05:49:43.313737080 -0600
+++ installer/misc/mkpackage.py	2014-05-05 20:00:00.000000000 -0600
@@ -27,7 +27,7 @@ libDir2     = libDir + '/csound/lib'
 # single precision plugin libraries
 pluginDir32 = libDir + '/csound/plugins'
 # double precision plugin libraries
-pluginDir64 = libDir + '/csound/plugins64'
+pluginDir64 = libDir + '/csound/plugins'
 # documentation
 docDir      = instPrefix + '/share/doc/csound'
 # tclcsound.so
--- Top/csmodule.c.orig	2014-05-04 05:49:41.341718675 -0600
+++ Top/csmodule.c	2014-05-05 20:00:00.000000000 -0600
@@ -142,9 +142,9 @@ static  const   char    *plugindir64_env
 #  define ENABLE_OPCODEDIR_WARNINGS 0
 #  ifndef CS_DEFAULT_PLUGINDIR
 #    ifndef USE_DOUBLE
-#      define CS_DEFAULT_PLUGINDIR  "/usr/local/lib/csound/plugins"
+#      define CS_DEFAULT_PLUGINDIR  "/usr/lib/csound/plugins"
 #    else
-#      define CS_DEFAULT_PLUGINDIR  "/usr/local/lib/csound/plugins64"
+#      define CS_DEFAULT_PLUGINDIR  "/usr/lib64/csound/plugins"
 #    endif
 #  endif
 #endif
