--- csound5/SConstruct.enable-fluidsynth	2007-08-21 07:27:54.000000000 -0400
+++ csound5/SConstruct	2007-08-22 22:38:16.000000000 -0400
@@ -221,6 +221,9 @@ commandOptions.Add('buildNewParser',
 commandOptions.Add('buildvst4cs',
     'Set to 1 to build vst4cs plugins (requires Steinberg VST headers)',
     '0')
+commandOptions.Add('useFluidsynth',
+    'Set to 1 if you want Fluidsynth support',
+    '0')
 
 # Define the common part of the build environment.
 # This section also sets up customized options for third-party libraries, which
@@ -478,6 +481,7 @@ if not tclhfound:
         tmp = '%s/tcl.h' % i
         tclhfound = tclhfound or configure.CheckHeader(tmp, language = "C")
 luaFound = configure.CheckHeader("lua.h", language = "C")
+fluidsynthFound = configure.CheckHeader("fluidsynth.h", language = "C")
 swigFound = 'swig' in commonEnvironment['TOOLS']
 print 'Checking for SWIG... %s' % (['no', 'yes'][int(swigFound)])
 pythonFound = configure.CheckHeader("Python.h", language = "C")
@@ -1353,7 +1357,7 @@ else:
 
 # FLUIDSYNTH OPCODES
 
-if not configure.CheckHeader("fluidsynth.h", language = "C"):
+if not (commonEnvironment['useFluidsynth'] == '1' and fluidsynthFound):
     print "CONFIGURATION DECISION: Not building fluid opcodes."
 else:
     print "CONFIGURATION DECISION: Building fluid opcodes."
