--- InOut/CMakeLists.txt.orig	2014-05-04 05:49:41.223717574 -0600
+++ InOut/CMakeLists.txt	2014-05-05 21:00:00.000000000 -0600
@@ -45,10 +45,6 @@ if(USE_PORTMIDI)
     find_library(PORTMIDI_LIBRARY portmidi)
 #    include_directories(${PORTMIDI_INCLUDE_PATH})
 #    include_directories(${PORTTIME_INCLUDE_PATH})
-    if(WIN32)
-    else()
-        find_library(PORTTIME_LIBRARY porttime)
-    endif()
 endif()
 if(USE_JACK)
     find_library(JACK_LIBRARY jack)
