--- Csound5.19.01/install.py.orig	2013-08-07 15:17:29.838998821 +0100
+++ Csound5.19.01/install.py	2013-08-07 15:18:40.184697145 +0100
@@ -288,7 +288,7 @@
 #err = installFile('opcodes.dir', pluginDir)
 installErrors = installErrors or err
 pluginList = findFiles('.', 'lib[A-Za-z].*\\.so')
-for i in ['libcsound.so', 'libcsound64.so']:
+for i in ['libcsound.so', 'libcsound64.so', 'libcsnd.so']:
     if i in pluginList:
         pluginList.remove(i)
 for i in pluginList:
@@ -305,6 +305,7 @@
 
 print ' === Installing language interfaces ==='
 wrapperList = [['csnd\\.py', '0', pythonDir],
+               ['_csnd\\.so', '1', pythonDir],
                ['loris\\.py', '0', pythonDir],
                ['CsoundVST\\.py', '0', pythonDir],
                ['CsoundVSTCsoundAC\\.py', '0', pythonDir],
