--- Csound5.10.1/install.py.orig	2009-03-31 11:31:45.000000000 +0100
+++ Csound5.10.1/install.py	2009-03-31 11:32:01.000000000 +0100
@@ -152,7 +152,7 @@
     fileList += [fileName]
     fullName = concatPath([instDir, fileName])
     err = runCmd(['install', '-p', '-m', perm, src, fullName])
-    if err == 0 and stripMode != '':
+    if err == 0 and stripMode == 'packman':
         err = runCmd(['strip', stripMode, fullName])
     if err == 0:
         addMD5(fullName, fileName)
@@ -333,7 +333,7 @@
   for i in xmgList:
     makeDir(concatPath([xmgDir, i, 'LC_MESSAGES']))
     src = 'po/' + i + '/LC_MESSAGES/csound5.mo'
-    fileName = concatPath([xmgDir, i, 'LC_MESSAGES/csound5.mo'])
+    fileName = concatPath([instDir, xmgDir, i, 'LC_MESSAGES/csound5.mo'])
     err = runCmd(['install', '-p', '-m', '0644', src, fileName])
     if err == 0:
         addMD5(fileName, fileName)
