--- Csound5.03.0/SConstruct.enable-fluidsynth	2006-10-21 11:03:22.000000000 -0400
+++ Csound5.03.0/SConstruct	2006-10-23 15:36:10.000000000 -0400
@@ -225,6 +225,9 @@
     '0')
 opts.Add('isWinNT', 'On windows, if win2000 or NT is used',
           '%d' % withMSVC())
+opts.Add('useFluidsynth',
+    'Set to 1 if you want OSC support',
+    '0')
 
 # Define the common part of the build environment.
 # This section also sets up customized options for third-party libraries, which
@@ -426,6 +429,7 @@
 pdhfound = configure.CheckHeader("m_pd.h", language = "C")
 tclhfound = configure.CheckHeader("tcl.h", language = "C")
 luaFound = configure.CheckHeader("lua.h", language = "C")
+fluidsynthFound = configure.CheckHeader("fluidsynth.h", language = "C")
 swigFound = 'swig' in commonEnvironment['TOOLS']
 print 'Checking for SWIG... %s' % (['no', 'yes'][int(swigFound)])
 pythonFound = configure.CheckHeader("Python.h", language = "C")
@@ -1228,7 +1232,7 @@
 
 # FLUIDSYNTH OPCODES
 
-if not configure.CheckHeader("fluidsynth.h", language = "C"):
+if not (commonEnvironment['useFluidsynth'] == '1' and fluidsynthFound):
     print "CONFIGURATION DECISION: Not building fluid opcodes."
 else:
     print "CONFIGURATION DECISION: Building fluid opcodes."
