--- cmake/CompilerOptimizations.cmake.orig	2014-05-04 05:49:42.169726403 -0600
+++ cmake/CompilerOptimizations.cmake	2014-05-05 20:45:00.000000000 -0600
@@ -19,30 +19,6 @@ if (HAS_CXX_FAST_MATH)
 endif()
 
 
-if(NOT "${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
-
-check_c_compiler_flag(-mfpmath=sse HAS_FPMATH_SSE)
-check_cxx_compiler_flag(-mfpmath=sse HAS_CXX_FPMATH_SSE)
-  if (HAS_FPMATH_SSE)
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mfpmath=sse")
-endif()
-if (HAS_CXX_FPMATH_SSE)
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mfpmath=sse")
-endif()
-
-endif()
-
-
-check_c_compiler_flag(-msse2 HAS_SSE2)
-check_cxx_compiler_flag(-msse2 HAS_CXX_SSE2)
-  if (HAS_SSE2)
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -msse2")
-endif()
-if (HAS_CXX_SSE2)
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -msse2")
-endif()
-
-
 check_c_compiler_flag(-fomit-frame-pointer HAS_OMIT_FRAME_POINTER)
 check_cxx_compiler_flag(-fomit-frame-pointer HAS_CXX_OMIT_FRAME_POINTER)
 if (HAS_OMIT_FRAME_POINTER)
