--- Csound5.19.01/Top/csmodule.c.orig	2013-08-07 15:25:36.964759575 +0100
+++ Csound5.19.01/Top/csmodule.c	2013-08-07 15:29:48.653836472 +0100
@@ -145,17 +145,19 @@
 #  endif
 #elif !(defined(_CSOUND_RELEASE_) && (defined(LINUX) || defined(__MACH__)))
 #  define ENABLE_OPCODEDIR_WARNINGS 1
-#  ifdef CS_DEFAULT_PLUGINDIR
-#    undef CS_DEFAULT_PLUGINDIR
-#  endif
-#  define CS_DEFAULT_PLUGINDIR      "."
+#  ifndef CS_DEFAULT_PLUGINDIR
+#    ifndef USE_DOUBLE
+#      define CS_DEFAULT_PLUGINDIR  "/usr/lib/csound/plugins"
+#    else
+#      define CS_DEFAULT_PLUGINDIR  "/usr/lib64/csound/plugins"
+#    endif
 #else
 #  define ENABLE_OPCODEDIR_WARNINGS 0
 #  ifndef CS_DEFAULT_PLUGINDIR
 #    ifndef USE_DOUBLE
-#      define CS_DEFAULT_PLUGINDIR  "/usr/local/lib/csound/plugins"
+#      define CS_DEFAULT_PLUGINDIR  "/usr/lib/csound/plugins"
 #    else
-#      define CS_DEFAULT_PLUGINDIR  "/usr/local/lib/csound/plugins64"
+#      define CS_DEFAULT_PLUGINDIR  "/usr/lib64/csound/plugins"
 #    endif
 #  endif
 #endif
