--- Csound5.03.0/SConstruct.enable-fluidsynth	2006-11-01 09:50:30.000000000 -0500
+++ Csound5.03.0/SConstruct	2006-11-01 10:36:08.000000000 -0500
@@ -224,6 +224,9 @@
     '0')
 opts.Add('isWinNT', 'On windows, if win2000 or NT is used',
           '%d' % withMSVC())
+opts.Add('useFluidsynth',
+    'Set to 1 if you want OSC support',
+    '0')
 opts.Add('buildCSEditor',
     'Set to 1 to build the Csound syntax highlighting text editor. Requires FLTK headers and libs',
     '0')
@@ -437,6 +440,7 @@
         tmp = '%s/tcl.h' % i
         tclhfound = tclhfound or configure.CheckHeader(tmp, language = "C")
 luaFound = configure.CheckHeader("lua.h", language = "C")
+fluidsynthFound = configure.CheckHeader("fluidsynth.h", language = "C")
 swigFound = 'swig' in commonEnvironment['TOOLS']
 print 'Checking for SWIG... %s' % (['no', 'yes'][int(swigFound)])
 pythonFound = configure.CheckHeader("Python.h", language = "C")
@@ -1263,7 +1267,7 @@
 
 # FLUIDSYNTH OPCODES
 
-if not configure.CheckHeader("fluidsynth.h", language = "C"):
+if not (commonEnvironment['useFluidsynth'] == '1' and fluidsynthFound):
     print "CONFIGURATION DECISION: Not building fluid opcodes."
 else:
     print "CONFIGURATION DECISION: Building fluid opcodes."
