--- Csound5.10.1/Top/csmodule.c.orig	2009-03-31 00:09:56.000000000 +0100
+++ Csound5.10.1/Top/csmodule.c	2009-03-31 00:21:59.000000000 +0100
@@ -126,17 +126,20 @@
 #else
 #if !(defined(_CSOUND_RELEASE_) && (defined(LINUX) || defined(__MACH__)))
 #  define ENABLE_OPCODEDIR_WARNINGS 1
-#  ifdef CS_DEFAULT_PLUGINDIR
-#    undef CS_DEFAULT_PLUGINDIR
+#  ifndef CS_DEFAULT_PLUGINDIR
+#    ifndef USE_DOUBLE
+#      define CS_DEFAULT_PLUGINDIR  "/usr/lib/csound/plugins"
+#    else
+#      define CS_DEFAULT_PLUGINDIR  "/usr/lib64/csound/plugins"
+#    endif
 #  endif
-#  define CS_DEFAULT_PLUGINDIR      "."
 #else
 #  define ENABLE_OPCODEDIR_WARNINGS 0
 #  ifndef CS_DEFAULT_PLUGINDIR
 #    ifndef USE_DOUBLE
-#      define CS_DEFAULT_PLUGINDIR  "/usr/local/lib/csound/plugins"
-#    Else
-#      define CS_DEFAULT_PLUGINDIR  "/usr/local/lib/csound/plugins64"
+#      define CS_DEFAULT_PLUGINDIR  "/usr/lib/csound/plugins"
+#    else
+#      define CS_DEFAULT_PLUGINDIR  "/usr/lib64/csound/plugins"
 #    endif
 #  endif
 #endif
