--- Csound5.12.1/SConstruct.orig	2010-12-27 00:07:56.497883757 +0000
+++ Csound5.12.1/SConstruct	2010-12-27 00:18:35.582892597 +0000
@@ -139,8 +139,8 @@
     'Set to 1 to build Python opcodes',
     '0')
 commandOptions.Add('prefix',
-    'Base directory for installs. Defaults to /usr/local.',
-    '/usr/local')
+    'Base directory for installs. Defaults to /usr.',
+    '/usr')
 commandOptions.Add('instdir',
     'For the install target: puts instdir before the prefix',
     '')
@@ -484,9 +484,9 @@
     if getPlatform() == 'sunos':
         commonEnvironment.Prepend(LIBPATH = ['.', '#.', '/lib/64', '/usr/lib/64'])
     else:    
-        commonEnvironment.Prepend(LIBPATH = ['.', '#.', '/usr/local/lib64'])
+        commonEnvironment.Prepend(LIBPATH = ['.', '#.'])
 else:
-    commonEnvironment.Prepend(LIBPATH = ['.', '#.', '/usr/local/lib'])
+    commonEnvironment.Prepend(LIBPATH = ['.', '#.'])
 
 if commonEnvironment['Word64'] == '1':
     if compilerSun():
@@ -506,8 +506,6 @@
 if getPlatform() == 'linux':
     commonEnvironment.Append(CCFLAGS = "-DLINUX")
     commonEnvironment.Append(CPPFLAGS = ['-DHAVE_SOCKETS'])
-    commonEnvironment.Append(CPPPATH = '/usr/local/include')
-    commonEnvironment.Append(CPPPATH = '/usr/include')
     commonEnvironment.Append(CPPPATH = '/usr/include')
     commonEnvironment.Append(CPPPATH = '/usr/X11R6/include')
     commonEnvironment.Append(CCFLAGS = "-DPIPES")
@@ -571,18 +569,17 @@
 
 if getPlatform() == 'linux':
     path1 = '/usr/include/python%s' % commonEnvironment['pythonVersion']
-    path2 = '/usr/local/include/python%s' % commonEnvironment['pythonVersion']
-    pythonIncludePath = [path1, path2]
+    pythonIncludePath = [path1]
     path1 = '/usr/include/tcl%s' % commonEnvironment['tclversion']
     path2 = '/usr/include/tk%s' % commonEnvironment['tclversion']
     tclIncludePath = [path1, path2]
     pythonLinkFlags = []
     if commonEnvironment['Lib64'] == '1':
         tmp = '/usr/lib64/python%s/config' % commonEnvironment['pythonVersion']
-        pythonLibraryPath = ['/usr/local/lib64', '/usr/lib64', tmp]
+        pythonLibraryPath = ['/usr/lib64', tmp]
     else:
         tmp = '/usr/lib/python%s/config' % commonEnvironment['pythonVersion']
-        pythonLibraryPath = ['/usr/local/lib', '/usr/lib', tmp]
+        pythonLibraryPath = ['/usr/lib', tmp]
     pythonLibs = ['python%s' % commonEnvironment['pythonVersion']]
 elif getPlatform() == 'sunos':
     path1 = '/usr/include/python%s' % commonEnvironment['pythonVersion']
@@ -2817,5 +2814,5 @@
 
 ###Code to create pkconfig files
 #env = Environment(tools=['default', 'scanreplace'], toolpath=['tools'])
-#env['prefix'] = '/usr/local'
+#env['prefix'] = '/usr'
 #env.ScanReplace('csound5.pc.in')
