--- Csound5.03.0/Opcodes/biquad.c.foo	2006-08-18 21:34:10.000000000 -0400
+++ Csound5.03.0/Opcodes/biquad.c	2006-08-18 21:35:24.000000000 -0400
@@ -118,7 +118,7 @@
     MYFLT *out, *in;
     double xn;
     MYFLT *fcoptr, *resptr;
-    double fco, res, fcon, kp, pp1d2, scale, k;
+    double fco, res, fcon, kp = 0, pp1d2 = 0, scale, k = 0;
     MYFLT max = p->maxint;
     double xnm1 = p->xnm1, y1nm1 = p->y1nm1, y2nm1 = p->y2nm1, y3nm1 = p->y3nm1;
     double y1n  = p->y1n, y2n = p->y2n, y3n = p->y3n, y4n = p->y4n;
@@ -194,8 +194,8 @@
     int n, nsmps = csound->ksmps;
     MYFLT *out, *fcoptr, *rezptr, *in;
     double fco, rez, xn, yn;
-    double fqcadj, c, rez2, a,
-          csq, b, tval; /* Temporary varibles for the filter */
+    double fqcadj, c, rez2 = 0, a = 0,
+          csq = 0, b = 0, tval = 0; /* Temporary varibles for the filter */
     double xnm1 = p->xnm1, xnm2 = p->xnm2, ynm1 = p->ynm1, ynm2 = p->ynm2;
 
     in     = p->in;
@@ -1029,7 +1029,7 @@
     MYFLT *fcoptr, *resptr, *distptr, *asymptr;
     MYFLT fco, res, dist, asym;
     MYFLT y = p->y, y1 = p->y1, y2 = p->y2;
-    MYFLT ih, fc, fco1, q, q1, fdbk, d, ad;
+    MYFLT ih, fc = 0, fco1, q = 0, q1 = 0, fdbk, d, ad;
 
     ih  = FL(0.001); /* ih is the incremental factor */
 
@@ -1098,7 +1098,7 @@
     long n, nsmps = csound->ksmps;
     MYFLT *out, *fcoptr, *rezptr, *in;
     double fco, rez, xn, yn;
-    double sin2, cos2, beta, alpha, gamma, mu, sigma, chi;
+    double sin2, cos2 = 0, beta = 0, alpha, gamma = 0, mu, sigma, chi;
     double theta;
     double xnm1 = p->xnm1, xnm2 = p->xnm2, ynm1 = p->ynm1, ynm2 = p->ynm2;
 
