--- Csound5.19.01/Top/csound.c.orig	2013-08-07 15:42:25.156102769 +0100
+++ Csound5.19.01/Top/csound.c	2013-08-07 15:43:43.659998379 +0100
@@ -1060,7 +1060,7 @@
 #ifdef __HAIKU__
       strcpy(s, "haiku");
 #else
-      strcpy(s, "PortAudio");
+      strcpy(s, "pulse");
 #endif
       csoundCreateConfigurationVariable(p, "rtaudio", s, CSOUNDCFG_STRING,
                                         0, NULL, &max_len,
